src/Modules/hilbert.jl

```diff
@@ -160,19 +168,43 @@ julia> den
 
 ```
 """
-function hilbert_series(SubM::SubquoModule{T}; parent::Union{Nothing,Ring} = nothing, backend::Symbol = :Abbott)  where T <: MPolyRingElem
+function hilbert_series(
+    SubM::SubquoModule{T}; 
+    parent::Union{Nothing,Ring} = hilbert_series_parent(base_ring(SubM)), 
+    backend::Symbol = :Abbott
+  )  where T <: MPolyRingElem
   @req is_z_graded(base_ring(SubM)) "ring must be ZZ-graded; use `multi_hilbert_series` otherwise"
   if parent === nothing
-    parent, _ = laurent_polynomial_ring(ZZ, :t)
+    parent, _ = laurent_polynomial_ring(ZZ, :t; cached=false)
   end
   HS, _ = multi_hilbert_series(SubM; parent=parent, backend=backend)
   return HS
 end
 
-function hilbert_series(F::FreeMod{T}; parent::Union{Nothing,Ring} = nothing, backend::Symbol = :Abbott)  where T <: MPolyRingElem
+function hilbert_series(
+    F::FreeMod{T}; 
+    parent::Union{Nothing,Ring} = hilbert_series_parent(base_ring(F)), 
+    backend::Symbol = :Abbott
+  )  where T <: MPolyRingElem
   @req is_z_graded(base_ring(F)) "ring must be ZZ-graded; use `multi_hilbert_series` otherwise"
   if parent === nothing
-    parent, _ = laurent_polynomial_ring(ZZ, :t)
+    parent, _ = laurent_polynomial_ring(ZZ, :t; cached=false)
   end
   return hilbert_series(sub_object(F,gens(F)); parent=parent, backend=backend)
 end
+
+function hilbert_series_parent(S::MPolyDecRing)
+  if !isdefined(S, :hilbert_series_parent)
+    if is_z_graded(S)
+      S.hilbert_series_parent = laurent_polynomial_ring(ZZ, :t; cached=false)[1]
```

Comment:
How about always using a multivariate polynomial ring?
```suggestion
      S.hilbert_series_parent = laurent_polynomial_ring(ZZ, [:t]; cached=false)[1]
```
That would make the return value of this function potentially type stable (well, with some further changes). As @jankoboehm says "this might also help with some other issues". But of course it might also cause issues... So this is really an open question.

Reply:
Unless people come up with qualified opinions on this, I will leave it as is since these were the rings used in the source code up to this point for the different cases. 

Type stability did not seem to matter before. If it matters now, I need clear advice which ring to use.